src/ape_node/provider.py

```diff
@@ -405,7 +409,10 @@ def restore(self, snapshot_id: SnapshotID):
             logger.error("Unable to set head to future block.")
             return
 
-        self.make_request("debug_setHead", [block_number_hex_str])
+        try:
+            self.make_request("debug_setHead", [block_number_hex_str])
+        except Exception:
```

Comment:
```suggestion
        except NotImplementedError:
```

Reply:
Perhaps we could re-raise the exception and also raise during `not self._supports_debug_set_head:` and then catch `NotImplementedError` in `chain.snapshot` method instead

Reply:
or perhaps the handling should just happen in he pytest integration